src/crimson/os/seastore/onode_manager/staged-fltree/fltree_onode_manager.cc

```diff
@@ -41,6 +42,10 @@ FLTreeOnodeManager::get_or_create_onode_ret
 FLTreeOnodeManager::get_or_create_onode(
   Transaction &trans,
   const ghobject_t &hoid) {
+  if (hoid.hobj.oid.name.length() + hoid.hobj.nspace.length()
+      > key_view_t::MAX_NS_OID_LENGTH) {
```

Comment:
@cyx1231st This is something we plan on fixing, right?  There are osd level limits on namespace and name length, but other than that this implementation must deal with any length.

Reply:
Yes, as long as we can figure out the upper-bound sizes of keys and values, we can then adjust the sizes of internal nodes and leaf nodes statically.

Dealing with **any length** of value without upper-bound is tricky I think. At least it seems unintuitive to me if node sizes need to be dynamic and are inhomogeneous at the same level.

Reply:
osd_max_object_name_len (2k by default) and object_max_object_namespace_len (256) are the configs.  We must handle any hobject within those limits.

Reply:
These means the key has upperbound, so they should be fine.

How about the imbedded value (`onode_layout_t`)? Does it also has upperbound?

Reply:
Yes, though we'll want it to be large enough to contain OI_ATTR and SS_ATTR for normal rbd blocks (with an xattr btree for other attrs or if those are too large).

Reply:
OK, this should be fine I believe.